### ISSUE TITLE
POINT - POINT_VEC3:NewFromVec2 not recognizing height parameter

### ISSUE THREAD

**Issue by Delta-99** [FlightControl-Master/MOOSE#550]
Using the method: POINT_VEC3:NewFromVec2(Vec2, LandHeightAdd) does not seem to recognize the LandHeightAdd parameter.

No matter what height I give it, it looks to be generating a POINT_VEC3 at the ground location.

**Comment by FlightControl-Master** [comment]
Ok! I'll fix asap!

**Comment by FlightControl-Master** [comment]
@Delta-99 I have been looking into this problem, but cannot find the problem!

**Comment by FlightControl-Master** [comment]
@Delta-99 Look at this piece of code:

```
  function COORDINATE:NewFromVec2( Vec2, LandHeightAdd ) 

    local LandHeight = land.getHeight( Vec2 )
    
    LandHeightAdd = LandHeightAdd or 0
    LandHeight = LandHeight + LandHeightAdd

    local self = self:New( Vec2.x, LandHeight, Vec2.y ) -- #COORDINATE

    self:F2( self )

    return self

  end
```

This should work, no?
When called from here:

```
  function POINT_VEC3:NewFromVec2( Vec2, LandHeightAdd )

    local self = BASE:Inherit( self, COORDINATE:NewFromVec2( Vec2, LandHeightAdd ) ) -- Core.Point#POINT_VEC3
    self:F2( self )

    return self
  end
```

Could you give a bit more details about the issue you have pls?
And was the issue on the master branch?

**Comment by Delta-99** [comment]
Hmmm, maybe I wasn't on the master branch. Probably not because of the issue the was introduced there rendering it impossible to use at the time. I don't recall seeing this inherit from Coordinate so if that is knew the code wasn't like that in the version I was looking at. HOWEVER, the code in the Coordinate function is exactly what I saw I think. Or something very very similar. It looked good to me but certainly wasn't working.

I likely don't have my code anywhere near the same anymore as I had to work around this and use SetY instead which is working fine for me.

When I get a chance I will try to change my code back to using this OR setup some simple example mission with the error. It might not be for a few days at least though.
